fairscale/optim/oss.py

```diff
@@ -103,8 +107,11 @@ def step(self, closure: Optional[Callable[[], float]] = None, **kwargs: Any) ->
         # Sync oss param_groups attributes in case they've been updated by a scheduler.
         self._sync_param_groups()
 
-        # Run the optimizer step on this shard only
-        loss = self.optim.step(closure=closure, **kwargs)  # type: ignore
+        # Run the optimizer step on this shard only:
+        if self._pass_closure:
+            loss = self.optim.step(closure=closure, **kwargs)  # type: ignore
```

Comment:
another option would be to check whether the closure is None or not, does not imply inspect, as you prefer